src/mpid/ch4/shm/ipc/gpu/gpu_post.c

```diff
@@ -191,6 +192,11 @@ int MPIDI_GPU_get_ipc_attr(const void *vaddr, int rank, MPIR_Comm * comm,
     ipc_attr->ipc_handle.gpu.offset = (uintptr_t) vaddr - (uintptr_t) pbase;
 
     ipc_attr->ipc_handle.gpu.global_dev_id = global_dev_id;
+
+    if (handle_obj == NULL) {
+        mpi_errno = MPIDI_GPU_ipc_handle_cache_insert(rank, comm, ipc_attr->ipc_handle.gpu);
+        MPIR_ERR_CHECK(mpi_errno);
+    }
```

Comment:
We have code insert cache in `ipc_p2p.h - MPIDI_IPCI_send_lmt`. If we do it in here, then we should remove it in `MPIDI_IPCI_send_lmt`. But what was the reason we need to do it here?

Reply:
Related to my comment above, if we make it to this part, an IPC handle is already created, so we should also add it to the cache now. Before, it could be created, but not meet the threshold requirement so it would not be cached and end up wasting resources.

I've removed the duplicate insertion in `MPIDI_IPCI_send_lmt`